opentok/src/audio_device.rs

```diff
@@ -0,0 +1,291 @@
+use crate::enums::{IntoResult, OtcBool, OtcError, OtcResult};
+
+use lazy_static::lazy_static;
+use std::collections::HashMap;
+use std::os::raw::c_void;
+use std::sync::{Arc, Mutex};
+
+// TODO: Get rid of INSTANCES. We need this currently because of some of the
+// macro calls remaining in this module.
+lazy_static! {
+    static ref INSTANCES: Arc<Mutex<HashMap<usize, AudioDevice>>> = Default::default();
+    static ref SINGLETON: Mutex<Option<AudioDevice>> = Mutex::new(None);
+}
+
+/// Initialize the global AudioDevice. This is called immediatetely after the
+/// library initialization. The AudioDevice has no associated callback by
+/// default. The application call `set_render_callbacks()` and/or
+/// `set_capture_callbacks()` to have the AudioDevice configured.
+pub fn initialize() -> OtcResult {
+    let audio_device = AudioDevice::new()?;
+    *SINGLETON.lock().unwrap() = Some(audio_device);
+    Ok(())
+}
+
+/// Associate render callbacks to the AudioDevice singleton. This should be
+/// called if the application wants to use a custom audio renderer.
+pub fn set_render_callbacks(callbacks: AudioDeviceCallbacks) {
```

Comment:
Maybe this should return a `Result<>`? Error out in case callbacks are set already?

Reply:
That sounds like a wise choice. Or at the very least warn about it if we want to allow overwriting callbacks.